FileDescriptor.cc

```diff
@@ -125,7 +126,7 @@ int FileDescriptor::readBlock(short relativeBlockNumber)
 
 int FileDescriptor::writeBlock(short relativeBlockNumber) 
 {
-	if(relativeBlockNumber >= IndexNode::MAX_FILE_BLOCKS)
+	if(relativeBlockNumber >= fileSystem->getBlockSize() + IndexNode::MAX_DIRECT_BLOCKS)
```

Comment:
I think that this needs to be `fileSystem->getBlockSize()/4` because each address is 4 bytes.
But this is a good solution.

Reply:
oh yes